### PR TITLE
[WIP] Stateful Helper Glimmer Support

### PR DESCRIPTION
**Pull request by al3x-edge**
Very rough, nasty, test-failing monster.  Just wanted to get your thoughts or suggestions on fixing some of the lookup errors.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: app/instance-initializers/ember-intl.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { filterBy } from 'ember-intl/utils/initialize';
 export function instanceInitializer(instance) {
     let service = instance.container.lookup('service:intl');
 
+    service.set('appInstance', instance);
```
Comment (jasonmit) [inline app/instance-initializers/ember-intl.js]: Not sure we need this, I'll do some investgating on my part

Comment (al3x-edge) [inline app/instance-initializers/ember-intl.js]: Aye. Felt dirty. Just not sure where else to get the reference for it. I think there is `this.container._registry` but being marked with an underscore and all I didn't investigate further.

Comment (jasonmit) [inline app/instance-initializers/ember-intl.js]: Why not `this.container.lookup`?

Comment (al3x-edge) [inline app/instance-initializers/ember-intl.js]: That's what it was before the change. I thought I was getting errors from it but maybe it was related to something else instead.

Comment (jasonmit) [inline app/instance-initializers/ember-intl.js]: #127 it all seems to be working except for tests and one other comment I pointed out which we need to guard against for backwards compat

**Comment by jasonmit** [comment]
Tests are now passing on #127 some clean up is needed before I merge it

**Pull request closed without merging**